### PR TITLE
[ts-sdk] Add an isValid function for AccountAddress

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Adds an isValid function

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/5127)

### CODE REVIEW

**Review by hariria — APPROVED** [review]
LGTM

**Review by davidiw — APPROVED** [review]

File: ecosystem/typescript/sdk/src/aptos_types/account_address.ts [hunk 1]
```diff
@@ -47,6 +47,29 @@ export class AccountAddress {
     return new AccountAddress(res);
   }
 
+  /**
+   * Checks if the string is a valid AccountAddress
+   * @param addr Hex string can be with a prefix or without a prefix,
+   *   e.g. '0x1aa' or '1aa'. Hex string will be left padded with 0s if too short.
+   */
+  static isValid(addr: MaybeHexString): boolean {
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_types/account_address.ts:55]: An empty string would pass in the current implementation, which is wrong.

**Review by jjleng — CHANGES_REQUESTED** [review]

File: ecosystem/typescript/sdk/src/aptos_types/account_address.ts (hunk 1 above)
Comment (davidiw) [reply ecosystem/typescript/sdk/src/aptos_types/account_address.ts:55]: Hmm... this is actually kinda interesting... isn't "" a compressed from of 0x0000...? I disagree :P

File: ecosystem/typescript/sdk/src/aptos_types/account_address.ts [hunk 2]
```diff
@@ -47,6 +47,29 @@ export class AccountAddress {
...
+      address = new HexString(`0${address.noPrefix()}`);
+    }
+
+    const addressBytes = address.toUint8Array();
+
+    if (addressBytes.length > AccountAddress.LENGTH) {
+      return false;
+    }
+
+    return true;
```
Comment (davidiw) [inline ecosystem/typescript/sdk/src/aptos_types/account_address.ts:70]: gah, you can just return from line 66

**Review by davidiw — APPROVED** [review]

File: ecosystem/typescript/sdk/src/aptos_types/account_address.ts (hunk 1 above)
Comment (gregnazario) [reply ecosystem/typescript/sdk/src/aptos_types/account_address.ts:56]: "" is not a compressed form of 0x000, the most compressed form would be 0x0", so I'm fixing this.

**Review by jjleng — APPROVED** [review]

File: ecosystem/typescript/sdk/src/aptos_types/account_address.ts [hunk 3]
```diff
@@ -47,6 +47,31 @@ export class AccountAddress {
     return new AccountAddress(res);
   }
 
+
+  /**
+   * Checks if the string is a valid AccountAddress
+   * @param addr Hex string can be with a prefix or without a prefix,
+   *   e.g. '0x1aa' or '1aa'. Hex string will be left padded with 0s if too short.
+   */
+  static isValid(addr: MaybeHexString): boolean {
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/aptos_types/account_address.ts:56]: might be good to add 
- check for 64 (or 66) chars long
- the string format, something like `/^(0[xX])?[a-fA-F0-9]{1,64}$/`

Comment (gregnazario) [reply ecosystem/typescript/sdk/src/aptos_types/account_address.ts:56]: It can be less than 64 characters, so that's not really to check right now

Comment (0xmaayan) [reply ecosystem/typescript/sdk/src/aptos_types/account_address.ts:56]: a valid account address is less than 64 chars?

Also, not really a fan of padding the address with leading 0s if its too short, that can potentially make an invalid address into a valid address which it shouldn't. for example, If someone types/copy-past it wrong and missing some chars, thats not a valid address and we should be blocking it.

**Pull request merged**